### PR TITLE
Fix/error msg extension

### PR DESCRIPTION
**Pull request by jbout**
Better error reporting of unsatisfied requirements

### CODE REVIEW

**Comment by gyszucs** [comment]
Is there a way to have this exception thrown somehow?

**Comment by jbout** [comment]
Add an extension which doesn't exist as required to one of your extensions

**Comment by siwane** [comment]
what do u think about auto install if the extension exists but not installed?

**Comment by jbout** [comment]
I worry that it would be "too magically" , especially in an AWS environment, where one of the web-servers would suddenly automatically install an extension

File: helpers/class.ExtensionHelper.php [hunk 1]
```diff
@@ -69,7 +69,11 @@ public static function sortByDependencies($extensions) {
                 }
             }
             if (count($unsorted) == $before) {
-                throw new common_exception_Error('Unable to resolve extension dependencies');
+                if (count($unsorted) == 1) {
+                    throw new common_exception_Error('Missing extensions '.implode(',', $missing).' for: '.implode(',', array_keys($unsorted)));
```
Comment (siwane) [inline helpers/class.ExtensionHelper.php]: Do u have a more elegant way to do that? ( ```if (count($unsorted) == 1 : implode(',', array_keys($unsorted)));```)

**Pull request merged**